File: client/Assets/Scripts/Message/info/chenliang/moba/message/FieldType.cs
namespace info.chenliang.moba.message{
public class FieldType{
public const int POSITION_X=0;
public const int POSITION_Z=1;
public const int DIRECTION=2;
public const int MAX_VALUE=3;
public static string ToString(byte value){
if(value == POSITION_X){
return "POSITION_X";
}
if(value == POSITION_Z){
return "POSITION_Z";
}
if(value == DIRECTION){
return "DIRECTION";
}
return null;
}
public static int ToValue(string name){
if(name == "POSITION_X"){
return POSITION_X;
}
if(name == "POSITION_Z"){
return POSITION_Z;
}
if(name == "DIRECTION"){
return DIRECTION;
}
return -1;
}
}
}

File: server/nobody/src/main/java/info/chenliang/moba/message/FieldType.java
package info.chenliang.moba.message;
public class FieldType{
public final static int ID=0;
public final static int POSITION_X=1;
public final static int POSITION_Z=2;
public final static int MAX_VALUE=3;
public static String toString(byte value){
if(value == ID){
return "ID";
}
if(value == POSITION_X){
return "POSITION_X";
}
if(value == POSITION_Z){
return "POSITION_Z";
}
return null;
}
public static int toValue(String name){
if(name.equals("ID")){
return ID;
}
if(name.equals("POSITION_X")){
return POSITION_X;
}
if(name.equals("POSITION_Z")){
return POSITION_Z;
}
return -1;
}
}
